File: application/entities/organisation.php
<?php
namespace entities;

/**
 * @Entity
 * */
class Organisation {

    //...
    /**
     * @Id @GeneratedValue @Column(type="integer")
     * @var integer
     */
    private $id = 0;
    //...
    /**
     * @OneToMany(targetEntity="Website", mappedBy="organisation")
     */
    private $websites = array();
    /**
     * @OneToMany(targetEntity="User", mappedBy="organisation")
     */
    private $users = array();
    /**
     * @OneToMany(targetEntity="User_group", mappedBy="organisation")
     */
    private $user_groups = array();
    /**
     * @OneToMany(targetEntity="Product", mappedBy="organisation")
     */
    private $products = array();
    /**
     * @OneToMany(targetEntity="Category", mappedBy="organisation")
     */
    private $categories = array();
    /**
     * @Column(type="string")
     * @var string
     */
    private $hash = "";
    /**
     * @Column(type="string")
     * @var string
     */
    private $created = "";
    /**
     * @Column(type="string")
     * @var string
     */
    public $name = "";
    /**
     * @Column(type="string")
     * @var string
     */
    private $company_reg = "";
    /**
     * @Column(type="string")
     * @var string
     */
    private $address = "";
    /**
     * @Column(type="string")
     * @var string
     */
    private $zip = "";
    /**
     * @Column(type="string")
     * @var string
     */
    private $city = "";
    /**
     * @Column(type="string")
     * @var string
     */
    private $email = "";
    /**
     * @Column(type="string")
     * @var string
     */
    private $telephone = "";
    /**
     * @Column(type="string")
     * @var string
     */
    private $tracking_code = "";
    /**
     * @Column(type="string")
     * @var string
     */
    private $tracking_code_invoice = "";


    public function get_id() {
        return $this->id;
    }

    public function set_id($id) {
        $this->id = $id;
    }

    public function get_users() {
        return $this->users;
    }

    public function set_users($users) {
        $this->users = $users;
    }

    public function get_user_groups() {
        return $this->user_groups;
    }

    public function set_user_groups($user_groups) {
        $this->user_groups = $user_groups;
    }

    public function get_products() {
        return $this->products;
    }

    public function set_products($products) {
        $this->products = $products;
    }

    public function get_categories() {
        return $this->categories;
    }

    public function set_categories($categories) {
        $this->categories = $categories;
    }

    public function get_hash() {
        return $this->hash;
    }

    public function set_hash($hash) {
        $this->hash = $hash;
    }

    public function get_created() {
        return $this->created;
    }

    public function set_created($created) {
        $this->created = $created;
    }

    public function get_name() {
        return $this->name;
    }

    public function set_name($name) {
        $this->name = $name;
    }

    public function get_company_reg() {
        return $this->company_reg;
    }

    public function set_company_reg($company_reg) {
        $this->company_reg = $company_reg;
    }

    public function get_address() {
        return $this->address;
    }

    public function set_address($address) {
        $this->address = $address;
    }

    public function get_zip() {
        return $this->zip;
    }

    public function set_zip($zip) {
        $this->zip = $zip;
    }

    public function get_city() {
        return $this->city;
    }

    public function set_city($city) {
        $this->city = $city;
    }

    public function get_email() {
        return $this->email;
    }

    public function set_email($email) {
        $this->email = $email;
    }

    public function get_telephone() {
        return $this->telephone;
    }

    public function set_telephone($telephone) {
        $this->telephone = $telephone;
    }

    public function get_tracking_code() {
        return $this->tracking_code;
    }

    public function set_tracking_code($tracking_code) {
        $this->tracking_code = $tracking_code;
    }

    public function get_tracking_code_invoice() {
        return $this->tracking_code_invoice;
    }

    public function set_tracking_code_invoice($tracking_code_invoice) {
        $this->tracking_code_invoice = $tracking_code_invoice;
    }

    ///// TO SOMEHOW MAINTAIN GOOD CODING PRACTICE WHILE SUPPORTING json_encode /////
    public function toArray() {
        return $this->processArray(get_object_vars($this));
    }

    private function processArray($array) {
        foreach ($array as $key => $value) {
            if (is_object($value)) {
                $array[$key] = $value->toArray();
            }
            if (is_array($value)) {
                $array[$key] = $this->processArray($value);
            }
        }
        // If the property isn't an object or array, leave it untouched
        return $array;
    }

    public function __toString() {
        return json_encode($this->toArray());
    }

    public function encodeJSON() {
        foreach ($this as $key => $value) {
            $json->$key = $value;
        }
        return json_encode($json);
    }

    public function decodeJSON($json_str) {
        $json = json_decode($json_str, 1);
        foreach ($json as $key => $value) {
            $this->$key = $value;
        }
    }

}

File: Shared/src/main/java/dk/apaq/shopsystem/model/Organisation.java
package dk.apaq.shopsystem.model;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;
import javax.persistence.Entity;
import javax.persistence.OneToMany;
import org.hibernate.annotations.Cascade;

/**
 * Specifies an organisation
 */
@Entity
public class Organisation extends AbstractEntity implements Serializable {

    @OneToMany
    private List<SystemUser> users = new ArrayList<SystemUser>();

    private String name;

    private String company_reg;

    private String address;

    private String zip;
    
    private String city;

    private String country = "US";
    
    private String email;
    
    private String telephone;

    private long initialOrdernumber=1;

    private long initialInvoiceNumber=1;

    private int defaultPaymentPeriodInDays=8;

    private String currency = "USD";

    public String getCountry() {
        return country;
    }

    public void setCountry(String country) {
        this.country = country;
    }

    public String getCurrency() {
        return currency;
    }

    public void setCurrency(String currency) {
        this.currency = currency;
    }

    public long getInitialInvoiceNumber() {
        return initialInvoiceNumber;
    }

    public void setInitialInvoiceNumber(long initialInvoiceNumber) {
        this.initialInvoiceNumber = initialInvoiceNumber;
    }

    public long getInitialOrdernumber() {
        return initialOrdernumber;
    }

    public void setInitialOrdernumber(long initialOrdernumber) {
        this.initialOrdernumber = initialOrdernumber;
    }

    public int getDefaultPaymentPeriodInDays() {
        return defaultPaymentPeriodInDays;
    }

    public void setDefaultPaymentPeriodInDays(int paymentPeriodInDays) {
        this.defaultPaymentPeriodInDays = paymentPeriodInDays;
    }

    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    public String getCity() {
        return city;
    }

    public void setCity(String city) {
        this.city = city;
    }

    public String getCompany_reg() {
        return company_reg;
    }

    public void setCompany_reg(String company_reg) {
        this.company_reg = company_reg;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getTelephone() {
        return telephone;
    }

    public void setTelephone(String telephone) {
        this.telephone = telephone;
    }

    public List<SystemUser> getUsers() {
        return users;
    }

    public void setUsers(List<SystemUser> users) {
        this.users = users;
    }

    public String getZip() {
        return zip;
    }

    public void setZip(String zip) {
        this.zip = zip;
    }

    

}
